### PR TITLE
MOB-1972 Update pull request template

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-1972]

## Context

In an ongoing effort to streamline our development workflow and enhance the quality of our codebase, this pull request introduces an improved pull request template. 
The template redefines how we submit and review pull requests, providing a more organized and structured approach for each contribution.

### Added

- Requires PR titles to follow the format "[MOB-####] PR TITLE." This structured format enhances traceability and cross-referencing.
- Introduces a "Before merging" checklist, ensuring that PRs undergo essential quality checks, including:
   -  real device and simulator testing
   - unit testing
   - localization updates
   - the addition of necessary comments, and documentation updates. 

This checklist emphasizes code quality and maintainability.
It's a small step towards an even more automated approach adding a linter utilizing for example the `Dangerfile` we have.

[MOB-1972]: https://ecosia.atlassian.net/browse/MOB-1972?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ecotopian — DISMISSED** [review]
Nice one! Once agreed I can bring this template to Android, too.

File: PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
-## Other
+## Other
+
+## Before merging
+
+### Checklist
+
+- [ ] I performed some relevant testing on a real device and/or simulator
+- [ ] I wrote Unit Tests that confirm the expected behaviour
+- [ ] I updated the Localization if needed
```
Comment (lucaschifino) [inline PULL_REQUEST_TEMPLATE:20]: Would it make sense to mention `(only the english source file since Transifex automation does the rest)`?

**Review by lucaschifino — DISMISSED** [review]
Looks good! Added a small non-blocking suggestion

File: PULL_REQUEST_TEMPLATE (hunk 1 above)
Comment (d4r1091) [reply PULL_REQUEST_TEMPLATE:20]: Awesome addition that provides even more clarity!

**Review by lucaschifino — APPROVED** [review]
LGTM 🟢

**Pull request merged**